### PR TITLE
CBG-210: Limit attachments to 20MB

### PR DESCRIPTION
**Pull request by JRascagneres**
- [x] Update manifest for walrus once PR merged https://github.com/couchbaselabs/walrus/pull/42
- [x] http://uberjenkins.sc.couchbase.com:8080/view/Build/job/sync-gateway-integration-master/958/

Also added sg_config.json to .gitignore in order to avoid committing local configs

### CODE REVIEW

File: db/crud.go [hunk 1]
```diff
@@ -990,6 +990,13 @@ func (db *Database) updateAndReturnDoc(
 			doc.History[newRevID].Channels = channelSet
 		}
 
+		//Need to check and add attachments here to ensure the attachment is within size constraints
+		attachmentErr := db.setAttachments(newAttachments)
+		if attachmentErr != nil{
+			err = errors.Wrap(attachmentErr, "Error adding attachment")
```
Comment (bbrks) [inline db/crud.go]: Could we create a HTTP error instead like the errors further up? That way we can set a more appropriate status code than 500 (like HTTP 413 Request Entity Too Large)

**Review by bbrks — CHANGES_REQUESTED** [review]

**Review by bbrks — DISMISSED** [review]
pending changes from updated walrus PR

File: db/crud.go [hunk 2]
```diff
@@ -17,8 +17,10 @@ import (
 	"strings"
 	"time"
 
+	"github.com/pkg/errors"
+
```
Comment (adamcfraser) [inline db/crud.go]: Can remove this whitespace and reformat.

File: rest/api_test.go [hunk 3]
```diff
@@ -12,8 +12,10 @@ package rest
 import (
 	"bytes"
 	"compress/gzip"
+	"encoding/base64"
 	"encoding/json"
 	"fmt"
+	"github.com/couchbaselabs/walrus"
```
Comment (adamcfraser) [inline rest/api_test.go]: Walrus should be down with the non-native library imports.

**Review by adamcfraser — CHANGES_REQUESTED** [review]
Looks good - just some minor cleanup on the imports.  Left a comment about moving setAttachments mainly for future reference.

File: .gitignore [hunk 4]
```diff
@@ -10,3 +10,4 @@
 /pkg
 *.exe
 *.iml
+sg_config.json
```
Comment (JFlath) [inline .gitignore:13]: Is this something we want in the main repo? Seems a little bit specific...

File: db/change_cache.go [hunk 5]
```diff
@@ -459,7 +459,7 @@ func (c *changeCache) DocChangedSynchronous(event sgbucket.FeedEvent) {
 		// No sync metadata found - check whether we're mid-upgrade and attempting to read a doc w/ metadata stored in xattr
 		migratedDoc, _ := c.context.checkForUpgrade(docID)
 		if migratedDoc != nil && migratedDoc.Cas == event.Cas {
-			base.Infof(base.KeyCache, "Found mobile xattr on doc %q without _sync property - caching, assuming upgrade in progress.",  base.UD(docID))
+			base.Infof(base.KeyCache, "Found mobile xattr on doc %q without _sync property - caching, assuming upgrade in progress.", base.UD(docID))
```
Comment (JFlath) [inline db/change_cache.go:462]: /nitpick - unrelated change

File: .gitignore (hunk 4 above)
Comment (JRascagneres) [reply .gitignore:13]: Discussed this with Adam yesterday. Ben and I basically have this as our SG config and I may have accidentally pushed it. This was added to avoid this.

File: db/change_cache.go (hunk 5 above)
Comment (JRascagneres) [reply db/change_cache.go:462]: gofmt did it

**Review by adamcfraser — APPROVED** [review]
